datajoint_utilities/dj_worker/__init__.py

```diff
@@ -225,20 +238,30 @@ def __init__(
         self._pipeline_modules = {}
 
     def __call__(self, process, **kwargs):
-        if isinstance(process, dj.user_tables.TableMeta):
-            self._processes_to_run.append(("dj_table", process, kwargs))
-            schema_name = process.full_table_name.split(".")[0].replace("`", "")
-            if schema_name not in self._pipeline_modules:
-                self._pipeline_modules[schema_name] = dj.create_virtual_module(
-                    schema_name, schema_name
-                )
+        if is_djtable(process):
+            self.add_table_process(process, **kwargs)
         elif inspect.isfunction(process) or inspect.ismethod(process):
-            self._processes_to_run.append(("function", process, kwargs))
+            self.add_function_process(process, **kwargs)
         else:
             raise NotImplemented(
                 f"Unable to handle processing step of type {type(process)}"
             )
 
+    def add_table_process(self, table, position_=None, **kwargs):
+        index = len(self._processes_to_run) if position_ is None else position_
+        schema_name = table.database
+        if not schema_name:
+            return
+        self._processes_to_run.insert(index, ("dj_table", table, kwargs))
+        if schema_name not in self._pipeline_modules:
+            self._pipeline_modules[schema_name] = dj.create_virtual_module(
+                schema_name, schema_name
+            )
+
+    def add_function_process(self, callable, position_=None, **kwargs):
+        index = len(self._processes_to_run) if position_ is None else position_
+        self._processes_to_run.insert(index, ("function", callable, kwargs))
+
```

Comment:
Can we combine these 2 functions into one - i.e. `add_process`, and detect/handle DJ table or callable accordingly, by checking using `is_djtable()` call?

Reply:
Yes I can if you're okay with adding the `position_` argument to the `__call__` method.